pkg/handler/alert_test.go

```diff
@@ -0,0 +1,37 @@
+package handler
+
+import (
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewAlertHandler(t *testing.T) {
+	require.NotEmpty(t, handlers.alertHandler)
+}
+
+func TestAlertRuleFiles(t *testing.T) {
+	alertHandler1 := handlers.alertHandler
+	r := gin.Default()
+	r.GET("/", alertHandler1.AlertRuleFiles)
+	w := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "/", nil)
+	r.ServeHTTP(w, req)
+	require.Equal(t, 200, w.Code)
+	want := `[{"commonLabels":{"rulefile":"sample-v3","severity":"silence"},"datasourceSelector":{"system":"","type":"prometheus"},"groups":[{"name":"sample","ruleAlerts":[{"rule":{"alert":"S00-AlwaysOn","expr":"vector(1234)","for":"0s","labels":{"hello":"world"},"annotations":{"summary":"AlwaysOn value={{ $value }}"}},"alerts":[{"datasource":{"type":"prometheus","name":"prometheus","url":"","basicAuth":false,"basicAuthUser":"","basicAuthPassword":"","isMain":true},"state":0,"activeAt":0}]},{"rule":{"alert":"S01-Monday","expr":"day_of_week() == 1 and hour() \u003c 2","for":"0s","annotations":{"summary":"Monday"}},"alerts":[{"datasource":{"type":"prometheus","name":"prometheus","url":"","basicAuth":false,"basicAuthUser":"","basicAuthPassword":"","isMain":true},"state":0,"activeAt":0}]},{"rule":{"alert":"S02-NewNamespace","expr":"time() - kube_namespace_created \u003c 120","for":"0s","annotations":{"summary":"labels={{ $labels }} namespace={{ $labels.namespace }} value={{ $value }}"}},"alerts":[{"datasource":{"type":"prometheus","name":"prometheus","url":"","basicAuth":false,"basicAuthUser":"","basicAuthPassword":"","isMain":true},"state":0,"activeAt":0}]}]}]}]`
+	require.Equal(t, want, w.Body.String())
+}
+
+func TestSendTestAlert(t *testing.T) {
+	alertHandler1 := handlers.alertHandler
+	r := gin.Default()
+	r.GET("/", alertHandler1.SendTestAlert)
+	w := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "/", nil)
+	r.ServeHTTP(w, req)
+	require.Equal(t, 200, w.Code)
+	want := `{"status":"success"}`
+	require.Equal(t, want, w.Body.String())
+}
```

Comment:
alertHandler1 은, TestSendTestAlert 와 TestAlertRuleFiles 2개 함수에서만 사용되는 것 같은데
alertHandler 변수와 용도가 다른지 여쭤보고 싶습니다.

변수 뒤의 의미가 약한 상수가 있을 경우, 추후 유지 보수 시에 혼동이 올 수 있어 여쭤봤습니다.
```suggestion
func TestSendTestAlert(t *testing.T) {
	alertHandler := handlers.alertHandler
	r := gin.Default()
	r.GET("/", alertHandler.SendTestAlert)
	w := httptest.NewRecorder()
	req := httptest.NewRequest("GET", "/", nil)
	r.ServeHTTP(w, req)
	require.Equal(t, 200, w.Code)
	want := `{"status":"success"}`
	require.Equal(t, want, w.Body.String())
}
```

Reply:
적절한 의미로 사용된 alertHandler1 일 경우 이 commit suggestion 은 기각하시면 될 것 같습니다.

Reply:
alertHandler1인 이유는, 단순히 alertHandler가 이미 구조체 type 이름으로 사용되고 있기 때문입니다.

https://github.com/kuoss/venti/blob/v0.2.3/pkg/handler/alert.go#L8-L10

이것에 관해서는... 둘 중 한쪽은 대문자로 시작하여 회피하는 방안(예: AlertHandler)도 논의된 바 있으나, export할 필요가 없는데 무리하게 그렇게 할 필요는 없다는 결론에 따라 저런 형태로 남았습니다.

참고로, 이와는 별개로, 패키지명과 겹치는 경우라면, 패키지명의 경우를 모두 소문자(예: alerthandler)로 하기로 논의된 바 있습니다.

Reply:
변수명 지정에 히스토리가 있었네요. 깔끔하게 설명해주셔서 잘 이해가 됐습니다.